src/UI/Implementation/Component/Input/Field/Checkbox.php

```diff
@@ -53,7 +53,11 @@ public function withValue($value)
             );
         }
 
-        return parent::withValue($value);
+        /**
+         * @var $clone Checkbox
+         */
+        $clone = parent::withValue($value);
+        return $clone;
```

Comment:
to tell the inspection (and the interested reader), that we passe back a checkbox in this context and that this can safely returned as checkbox (otherwise the injector and maybe also the reader will panic, since parent::withValue might return other things).

Reply:
Makes sense.